app/models/audio_file.rb

```diff
@@ -94,4 +99,21 @@ def set_status
       self.status_message = errors
     end
   end
+
+  def analyze!
```

Comment:
Would it make more sense to call this `process!` ... since it's doing both a copy and an analyze?